### PR TITLE
Features/pastushkov lexer optimization

### PR DESCRIPTION
**Pull request by opastushkov**

### CODE REVIEW

File: src/editor/codeeditor.cpp [hunk 1]
```diff
@@ -10,13 +10,16 @@
 #include<QMessageBox>
 #include<iostream>
 #include<QLabel>
+#include <QVector>
+
```
Comment (anantonyk) [inline src/editor/codeeditor.cpp:17]: remove blank line

File: src/editor/codeeditor.cpp [hunk 2]
```diff
@@ -78,18 +84,72 @@ CodeEditor::CodeEditor(QWidget *parent) : QPlainTextEdit(parent)
     fmtComment.setForeground(Qt::green);
     fmtUndefined.setUnderlineStyle(QTextCharFormat::WaveUnderline);
     fmtUndefined.setUnderlineColor(Qt::red);
-    fmtRegular.setForeground(Qt::black);
     fmtLiteral.setForeground(mConfigParam.mStringsColor);
     fmtKeyword.setForeground(mConfigParam.mBasicLiteralsColor);
     fmtComment.setForeground(mConfigParam.mCommentColor);
     fmtRegular.setForeground(mConfigParam.mCodeTextColor);
+    fmtRegular.setForeground(Qt::white);
```
Comment (anantonyk) [inline src/editor/codeeditor.cpp]: don`t hardcode colors! take it from settings

File: src/editor/codeeditor.cpp [hunk 3]
```diff
@@ -78,18 +84,72 @@ CodeEditor::CodeEditor(QWidget *parent) : QPlainTextEdit(parent)
...
+}
+
+void CodeEditor::handleLinesAddition(int changeStart, int lastLineWithChange, int lineDifference)
+{
+    QString changedCode;
+
+    if(lineDifference > 0)
```
Comment (anantonyk) [inline src/editor/codeeditor.cpp]: if (lineDifference > 0)
the same below

File: src/editor/codeeditor.cpp [hunk 4]
```diff
@@ -78,18 +84,72 @@ CodeEditor::CodeEditor(QWidget *parent) : QPlainTextEdit(parent)
...
     fmtUndefined.setUnderlineColor(Qt::red);
-    fmtRegular.setForeground(Qt::black);
     fmtLiteral.setForeground(mConfigParam.mStringsColor);
     fmtKeyword.setForeground(mConfigParam.mBasicLiteralsColor);
     fmtComment.setForeground(mConfigParam.mCommentColor);
     fmtRegular.setForeground(mConfigParam.mCodeTextColor);
+    fmtRegular.setForeground(Qt::white);
+}
+
+void CodeEditor::handleLinesAddition(int changeStart, int lastLineWithChange, int lineDifference)
```
Comment (anantonyk) [inline src/editor/codeeditor.cpp:144]: You can pass here a const or const& value to avoid some unnecessary changes
the same in function below

File: src/editor/codeeditor.cpp [hunk 5]
```diff
@@ -78,18 +84,72 @@ CodeEditor::CodeEditor(QWidget *parent) : QPlainTextEdit(parent)
...
+        {
+            mTokensList[i] = mLcpp->getTokens();
+        }
+    }
+}
+
+void CodeEditor::handleLinesDelition(int changeStart, int lastLineWithChange, int lineDifference)
+{
+    QString changedCode;
+    if(lineDifference < 0)
```
Comment (anantonyk) [inline src/editor/codeeditor.cpp]: unnecessary check

File: src/editor/keypressevents.cpp [hunk 6]
```diff
@@ -136,9 +136,9 @@ void EventSendLexem::operator()(CodeEditor * codeEditor, QKeyEvent *e)
     int position = cursor.position();
     for (const auto &it: editorTokens(codeEditor))
     {
-        if (it.mType == State::KW && it.mBegin <= position && it.mEnd >= position)
+        if (it[position].mType == State::KW && it[position].mBegin <= position && it[position].mEnd >= position)
```
Comment (PeterLysyk) [inline src/editor/keypressevents.cpp]: write every condition from new line. your code it's very obvious.
``` 
if (it[position].mType == State::KW
    && it[position].mBegin <= position 
    && it[position].mEnd >= position)
```

**Review by anantonyk — APPROVED** [review]

**Review by PeterLysyk — APPROVED** [review]
